addons/hr/models/hr_department.py

```diff
@@ -70,13 +70,13 @@ def _compute_master_department_id(self):
             department.master_department_id = int(department.parent_path.split('/')[0])
 
     def _compute_total_employee(self):
-        emp_data = self.env['hr.employee'].sudo()._read_group([('department_id', 'in', self.ids)], ['department_id'], ['__count'])
+        emp_data = self.env['hr.employee'].sudo()._read_group([('department_id', 'in', self.ids), ('company_id', 'in', self.env.companies.ids)], ['department_id'], ['__count'])
         result = {department.id: count for department, count in emp_data}
         for department in self:
             department.total_employee = result.get(department.id, 0)
 
     def _compute_plan_count(self):
-        plans_data = self.env['mail.activity.plan']._read_group([('department_id', 'in', self.ids)], ['department_id'], ['__count'])
+        plans_data = self.env['mail.activity.plan']._read_group([('department_id', 'in', self.ids), ('company_id', 'in', self.env.companies.ids)], ['department_id'], ['__count'])
```

Comment:
The plan can be set with an empty company field, those should be included as well
```suggestion
        plans_data = self.env['mail.activity.plan']._read_group([('department_id', 'in', self.ids), ('company_id', 'in', self.env.companies.ids + [False])], ['department_id'], ['__count'])
```